upload-api/stores/allocations.js

```diff
@@ -0,0 +1,211 @@
+import {
+  DynamoDBClient,
+  GetItemCommand,
+  PutItemCommand,
+  QueryCommand,
+} from '@aws-sdk/client-dynamodb'
+import { marshall, unmarshall } from '@aws-sdk/util-dynamodb'
+import { RecordKeyConflict, RecordNotFound } from '@web3-storage/upload-api/errors'
+import { base58btc } from 'multiformats/bases/base58'
+import * as Link from 'multiformats/link'
+
+/**
+ * @typedef {import('@web3-storage/upload-api/types').AllocationsStorage} AllocationsStorage
+ * @typedef {import('@web3-storage/upload-api/types').BlobAddInput} BlobAddInput
+ * @typedef {import('@web3-storage/upload-api/types').BlobListItem} BlobListItem
+ * @typedef {import('@web3-storage/upload-api/types').ListOptions} ListOptions
+ * @typedef {import('@web3-storage/upload-api/types').DID} DID
+ */
+
+/**
+ * Abstraction layer to handle operations on Store Table.
+ *
+ * @param {string} region
+ * @param {string} tableName
+ * @param {object} [options]
+ * @param {string} [options.endpoint]
+ * @returns {AllocationsStorage}
+ */
+export const createAllocationsStorage = (region, tableName, options = {}) => {
+  const dynamoDb = new DynamoDBClient({
+    region,
+    endpoint: options.endpoint,
+  })
+
+  return useAllocationsStorage(dynamoDb, tableName)
+}
+
+/**
+ * @param {DynamoDBClient} dynamoDb
+ * @param {string} tableName
+ * @returns {AllocationsStorage}
+ */
+export function useAllocationsStorage(dynamoDb, tableName) {
```

Comment:
This is quite similar with `storeTable` with few modifications given new columns